packages/react/src/ReactCreateEventComponent.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ * @flow
+ */
+
+import type {ReactEventComponent, ReactEventResponder} from 'shared/ReactTypes';
+
+import {REACT_EVENT_COMPONENT_TYPE} from 'shared/ReactSymbols';
+
+type EventComponentConfig = {
+  displayName: string,
+  props: null | Object,
+  responder: ReactEventResponder,
+};
+
+export function createEventComponent({
+  displayName,
+  props,
+  responder,
+}: EventComponentConfig): ReactEventComponent {
+  const eventComponent = {
+    $$typeof: REACT_EVENT_COMPONENT_TYPE,
+    displayName: displayName,
+    props: props,
+    responder: responder,
+  };
+  if (__DEV__) {
+    Object.freeze(eventComponent);
```

Comment:
Given we freeze here. We can remove the freeze operation in FiberComplete phase.